app/models/shop/stock.rb

```diff
@@ -0,0 +1,14 @@
+class Shop::Stock < ApplicationRecord
+  belongs_to :book
+  belongs_to :shop
+
+  validates :value, numericality: { greater_than_or_equal_to: 0 }
+
+  scope :with_books, -> (book_ids) { where(book_id: book_ids) }
+
+  def initialize(attrs = {})
+    super(attrs)
+    self.value = 0 if value.nil?
+  end
+
```

Comment:
rubocop won't like it :)

Reply:
Rubocop with default config see no offenses here.

Reply:
https://github.com/bbatsov/ruby-style-guide#empty-lines-around-bodies

Reply:
Heh, though it smth about initializer :( you're right, sorry.